finality-aleph/src/data_io.rs

```diff
@@ -388,18 +391,19 @@ pub(crate) async fn refresh_best_chain<B, BE, SC, C>(
                     .await
                     .expect("No best chain");
                 if *new_best_header.number() <= max_block_num {
-                    *proposed_block.lock() = AlephData::new(new_best_header.hash(), *new_best_header.number());
+                    *proposed_block.lock().await = AlephData::new(new_best_header.hash(), *new_best_header.number());
```

Comment:
First time see this concept that we assign to `lock()` on the left-hand side of assignment - so just double checking if this line is an equivalent to 
```rust
{
   let mut block = proposed_block.lock().await;                    
   *block = AlephData::new(new_best_header.hash(), *new_best_header.number());
}
```
as I'm not sure entirelely what's the lifetime of `MutexGourd` temporary object `*proposed_block.lock().await `.

Reply:
Yes, I'm pretty sure this is how it works. I mean I'm not sure about exact equivalence, as I'm not sure what this block compiles to. Anyway, lifetime of temporary object ends in this line and guard is released. And value is set.